### PR TITLE
Improve persistence and add test.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/full_client.toit [hunk 1]
```diff
@@ -489,29 +496,34 @@ class MemoryPersistenceStore implements PersistenceStore:
...
   do [block] -> none:
-    storage_.do block
+    storage_.do --values block
 
+  /** Whether no packet is persisted. */
+  is_empty -> bool:
+    return storage_.is_empty
+
+  /**
+  The count of persisted packets
```
Comment (kasperl) [inline src/full_client.toit:523]:
```suggestion
  The count of persisted packets.
```

File: tests/packet_test_client.toit [hunk 2]
```diff
@@ -45,7 +53,7 @@ with_packet_client create_transport/Lambda [block]
     idle.down
 
   task::
-    exception := catch --unwind=(on_handle_error == null):
+    exception := catch --unwind=(on_handle_error == null) --trace:
```
Comment (kasperl) [inline tests/packet_test_client.toit:56]: Do you want to keep the tracing turned on?

File: src/full_client.toit [hunk 3]
```diff
@@ -489,29 +496,34 @@ class MemoryPersistenceStore implements PersistenceStore:
...
-  get persistent_id/int -> PersistedPacket?:
-    return storage_.get persistent_id
+  get persisted_id/int -> PersistedPacket?:
+    return storage_.get persisted_id
 
-  remove persistent_id/int -> none:
-    storage_.remove persistent_id
+  remove_persisted_with_id persisted_id/int -> bool:
+    result := storage_.contains persisted_id
+    storage_.remove persisted_id
```
Comment (kasperl) [inline src/full_client.toit:508]: Maybe use `if_absent` parameter to return false? Seems better than having to hash and lookup twice.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/packet_test_client.toit (hunk 2 above)
Comment (floitsch) [reply tests/packet_test_client.toit:56]: Removed it. It's just in the test, but I don't think we need it.

File: src/full_client.toit (hunk 3 above)
Comment (floitsch) [reply src/full_client.toit:508]: Forgot that we had an `--if_absent`. Nice.
done.

**Pull request merged**